NitroxServer/ConsoleCommands/ExitCommand.cs

```diff
@@ -11,12 +13,66 @@ public ExitCommand() : base("exit", Optional<string>.Empty(), "Exits the server"
 
         public override void RunCommand(string[] args)
         {
+            Console.WriteLine("Stopping server:");
+            Console.WriteLine("Kicking all players...");
+            //TODO: Kicking all
+            Console.WriteLine("Players kicking failed. They'll kicked after server stop anyway.");
+            Console.WriteLine("Calling \"Save\" method...");
+            Console.WriteLine("Method called. Your anwser was: " + Save());
+            Console.WriteLine("Exiting...");
             Server.Instance.Stop();
+            Console.WriteLine("Exited.");
+            Console.WriteLine("Server stopped!");
         }
 
         public override bool VerifyArgs(string[] args)
         {
             return args.Length == 0;
         }
+        
+        bool Save()
+        {
+            Console.WriteLine("Save before exiting? [Y/N]");
+            char Output = Console.ReadKey(true).KeyChar;
+            if (Output == 'y')
+            {
+                Console.WriteLine("You pressed \"Y\" - Saving...");
+                Server.Instance.Save();
+                Console.WriteLine("Saved.");
+                Console.WriteLine("Exiting in 100 miliseconds.");
+                Thread.Sleep(100);
+                return true;
+            }
+            else if (Output == 'n')
+            {
+                return AreYouSure();
+            }
+            else
+            {
+                Console.WriteLine("Worng key. Calling \"Save\" method again...");
+                return Save();
+            }
+        }
+
+        bool AreYouSure()
+        {
+            Console.WriteLine("Are you sure? ALL UNSAVED PROGRESS WILL BE LOST! [Y/N]");
```

Comment:
I don't think it's a good idea to ask for console input in a command. These commands can be excuted in-game too and will break the server console. Instead I recommend just requiring a -y or -n argument to the command. If no arg is supplied you just tell the user to reenter the command with -y or -n.